### ISSUE TITLE
Modify default behaviour of map-factory

### ISSUE THREAD

**Issue by midknight41** [midknight41/map-factory#38]
The default behaviour of a mapper currently does a few things I don't like.

- It creates structure even when no source value is found
- It a transform is provided it will always call the transform.

I think the default behaviour should be:

- transforms are not called if the source value is not available
- structure is not created if the source value is not available

This is a breaking change so will be a semver major.

The existing behaviour will be supported by providing options to the ```createMapper()``` function.

```js

const options = {
  alwaysTransform: true,
  alwaysSet: true
};

const mapper = createMapper(options);
```

**Comment by midknight41** [comment]
@simranjeet2589 @phelanma You input would be appreciated.

**Comment by simranjeet2589** [comment]
Looks good to me.

**Comment by midknight41** [comment]
@Abhisheksharma01 Any concerns from your side of things?

**Comment by midknight41** [comment]
I've fixed the remaining issue on the v1 line and will now move onto finishing the work for v2.

**Comment by midknight41** [comment]
Consider the use case of a ```undefined``` or ```null``` source field mapping to a value array.
```js
const map = createMapper();
const src =  {
    item: null
  };

map("item").to("[]")
console.log(map.execute(src));
```
The current behaviour for this would return ```[]``` which still feels like the correct behaviour for v2. ```map.execute``` will always return an object of some kind but an empty object in this case feels weird. Any thoughts?

@simranjeet2589 
@Abhisheksharma01

**Comment by midknight41** [comment]
Lacking any feedback I'm going with the above approach.

**Comment by midknight41** [comment]
Published as v2.0.1. 

I screwed up with v2.0.0. 😭
